test/karma.config.js

```diff
@@ -0,0 +1,61 @@
+var pkg = require('../package.json');
+
+module.exports = function(karma) {
+
+  karma.set({
+    frameworks: ['browserify', 'mocha'],
+    files: ['api/*.test.js'],
+    preprocessors: {
+      'api/*.test.js': ['browserify']
+    },
+    browserify: {
+      debug: true,
+      transform: ['envify']
+    }
+  });
+
+  if (process.env.TRAVIS) {
+
+    if (!process.env.SAUCE_USERNAME || !process.env.SAUCE_ACCESS_KEY) {
+      process.stderr.write('SAUCE_USERNAME or SAUCE_ACCESS_KEY not set\n');
+      process.exit(1);
+    }
+
+    var customLaunchers = {
+      'SL_Chrome': {
+        base: 'SauceLabs',
+        browserName: 'chrome'
+      },
+      'SL_Firefox': {
+        base: 'SauceLabs',
+        browserName: 'firefox'
+      }
```

Comment:
I removed IE 11 here. Test failures due to lack of promises. I tried creating a small polyfill bootstrap file (`require('es6-promise').polyfil()`) to no avail. Surely it's doable but I want to dabble with that later.